packages/eslint-plugin-formatjs/rules/enforce-id.ts

```diff
@@ -45,9 +60,16 @@ function checkNode(context: Rule.RuleContext, node: TSESTree.Node) {
           }
         )
         if (id !== correctId) {
+          let message = `"id" does not match with hash pattern ${idInterpolationPattern}`
```

Comment:
sry 1 last thing shouldn't you move the regex check down to this block?
with that you can also specifically tell user which regex failed in the array

Reply:
Nope.
Whitelist provides different behavior - it allows bypass hash pattern check.
For example: this rule for some message enforces to be `id` equals 'ef127d'. Any other value will throw an error. So `idWhitelist: ['\\.'],` unlocks any custom string with dots, eg 'my.custom.id' - will be valid, but 'rfejkfnkjnnejkrnf' - invalid.

Reply:
Here I don't know which regex was matched and marks custom id as valid.

Here printing regexp in message error must inform users that they besides standard hash patter may use custom values for ids.

Reply:
yeah but the whitelist only matters if the IDs don't match right? which is what this block does.

Or the option is
```js
if (id === correctId) { return }
// do the rest
```

Reply:
you can also do a `.filter` instead of `.some` and it'll tell you which one matches

Reply:
> yeah but the whitelist only matters if the IDs don't match right? which is what this block does.

Nope. Whitelist allows using additional patterns for IDs that are written by a user.

In other words:
- if ID matches `idInterpolationPattern` – ✅ no error 
- if ID matches any regexp in `idWhitelist` – ✅ no error
- otherwise (no match to `idInterpolationPattern`  and `idWhitelist`) – 🛑 will be shown an error that ID does not match all patterns and regexps.